lib/_stream_writable.js

```diff
@@ -194,23 +194,16 @@ function writeAfterEnd(stream, cb) {
   process.nextTick(cb, er);
 }
 
-// If we get something that is not a buffer, string, null, or undefined,
-// and we're not in objectMode, then that's an error.
-// Otherwise stream chunks are all considered to be of length=1, and the
-// watermarks determine how many objects to keep in the buffer, rather than
-// how many bytes or characters.
```

Comment:
You should add back some comment here, explaining this change. `validChunk`  now only checks if a chunk is valid when it is not a `Buffer`.